Develop

yyy9942 opened the pull request:
* myBatis와 외부 DB를 연동하였습니다.
* application.properties에 profile을 적용하여 설정을 분리하였습니다.

Commit:
mybatis, mariadb 연동

Commit:
* Log 설정 삭제 - Logging 공부해서 다시 추가하겠습니다.
* DB, Mybatis 연결 설정
* 프로젝트 구조 설정

binaryyoung committed:
memberVO

컬럼 추가하였습니다.(test)

f-lab-dev commented:
마스터에 커밋하셨네요. 깃과 브랜치의 개념에 대해 더 공부해보시면 좋을 것 같습니다~
기본 구조에 대해서는 링크달아서 리뷰해드리겠습니다

f-lab-dev commented:
https://github.com/f-lab-edu/food-delivery/blob/develop/README.md

README.md는 의미있어야합니다. 이 프로젝트가 어떤 프로젝트인지 써주시면 좋을 것 같습니다.

f-lab-dev commented:
프로젝트의 파일들을 깔끔하게 유지하기 위해서 logs/spring-boot-logging.log 와 같이 프로젝트에 필요없는 부수적인 파일은 올리지 않아야합니다. 
`.gitignore`를 검색해보시고 이런 파일들은 삭제하는게 좋을 것 같습니다.

f-lab-dev commented:
resources 디렉토리에도 `.gitignore`가 있는데 최상위 디렉토리에 있는 한개로만 관리하는게 더 낫습니다.

f-lab-dev commented on src/main/java/com/delfood/config/DatabaseConfig.java:
```diff
@@ -0,0 +1,37 @@
+ 
+@Configuration
+@MapperScan(basePackages="com.delfood.dao")
+@EnableTransactionManagement
```
각 어노테이션을 여기에 붙여서 오는 효과가 무엇인지 알고계셔야합니다.
처음엔 주석으로 달아놓는 것도 괜찮은 방법입니다.

f-lab-dev commented on src/main/java/com/delfood/controller/MemberController.java:
```diff
@@ -0,0 +1,24 @@
+import com.delfood.vo.MemberVO;
+
+@RestController
+@RequestMapping("/member/")
```
보통 단수가 아닌 복수형태로 하는 것이 좋습니다.

f-lab-dev commented on src/main/java/com/delfood/controller/MemberController.java:
```diff
@@ -0,0 +1,24 @@
+	@Autowired
+	MemberService memberService;
+	
+	@GetMapping("list")
```
여기도 `/members`로 호출하면 리스트를 불러온다는걸 자연스럽게 추측할 수 있으므로 list라고 쓰지 않는 것이 좋습니다.
아래의 도큐먼트를 읽어보시는걸 추천합니다
https://docs.microsoft.com/ko-kr/azure/architecture/best-practices/api-design

f-lab-dev commented on src/main/java/com/delfood/dao/MemberMapper.java:
```diff
@@ -0,0 +1,12 @@
+
+@Repository
+public interface MemberMapper {
+	public List<MemberVO> getAll();
```
취향차이지만 보통 VO보다는 DTO라는 이름을 자주 씁니다. (Value Transfer Object의 줄임말입니다)

f-lab-dev commented on src/main/java/com/delfood/controller/MemberController.java:
```diff
@@ -0,0 +1,24 @@
+	MemberService memberService;
+	
+	@GetMapping("list")
+	public List<MemberVO> list() {
```
이 API는 어느 기능에서 사용하려고 하신건지 궁금합니다.
단순 전체 회원목록을 불러오는 기능은 어디에서도 쓰일 것 같지 않아서요.
각각의 API를 만드실 때 어느 곳에 필요한 기능인지 전부 염두에 두고 필요한 것만 만드셔야합니다.

f-lab-dev commented on src/main/java/com/delfood/service/MemberService.java:
```diff
@@ -0,0 +1,25 @@
+	 * @return
+	 * - 모든 멤버 리스트
+	 */
+	public List<MemberVO> getAll(){
```
여기도 필요하지 않은 기능인 것 같습니다.
1. 회원이 1억명이라면 1억명의 정보를 다 가져오게 됩니다. 이는 바로 장애의 원인이 됩니다.
2. 검색과 정렬조건이 없습니다. 특정한 회원이 아닌 임의의 회원목록을 가져오는 기능은 사용할 일이 없을 것 같습니다.

f-lab-dev commented on src/main/resources/mybatis/mapper/member.xml:
```diff
@@ -0,0 +1,9 @@
+<mapper namespace="com.delfood.dao.MemberMapper">
+
+    <select id="getAll" resultType="com.delfood.vo.MemberVO">
+        SELECT * FROM TB_MEMBER
```
1. `*`을 사용하는 것은 좋지 않습니다. 가져올 컬럼명을 정확하게 명시해주는 것이 좋습니다.
2. 위에도 달았지만 회원이 1억명이면 1억명의 정보를 다 가져오게 됩니다. 꼭 임의의 회원을 가져오고 싶으시다면 페이징처리를 하셔야합니다.

f-lab-dev reviewed (commented):
리뷰완료했습니다

yyy9942 committed:
Merge pull request #2 from f-lab-edu/feature/memberVO_setting

memberVO

Commit:
코드리뷰를 반영하였습니다!

필요하지 않은 기능인 멤버 리스트 불러오기는 삭제하였습니다.

Commit:
LOMBOK 추가

Commit:
VO에서 DTO로 명명 변경

Commit:
- log4j2를 아용하여 쿼리 로그와 result를 표시하도록 하였습니다.
- VO를 DTO로 변경하였습니다.
- lombok을 MemberDTO에 적용하였습니다.

Commit:
Bugfix
- DB와 매핑되지 않던 컬럼의 타입을 수정하였습니다.

Commit:
redis 설정

Commit:
Spring Session 의존성 추가

Commit:
- redis session 적용

Commit:
redis config에 serializer 변경
DAO => mapper로 명칭 변경
쿼리 id 명칭 변경

Commit:
profile에 따른 Logback 설정과 테스트.
redis 입출력 테스트

Commit:
.gitignore에서 log파일을 커밋하지 않도록 설정

Commit:
각 어노테이션, 라이브러리에 주석 추가

Commit:
회원가입과 로그인 구현

- 비밀번호 암호화(SHA 256)
비밀번호는 복호화를 할 필요없이 입력한 데이터와 일치여부만 확인하면 로그인이 가능하니 단방향 암호화 알고리즘 사용

- DTO를 DB와 맞게 수정

- 로그인, 회원가입 기능 추가

- 로그인, 회원가입 테스트

Commit:
ID 중복체크 구현

Commit:
비밀번호 수정 기능 추가

Commit:
readme 수정

Commit:
- HTTP Status 변경
- 고객 조회시 status 컬럼을 검사하여 delete일시 검색에서 제외하도록 설정
- 고객 삭제 구현

Commit:
- NoSuchAlgorithException을 RuntimeException으로 변환
- Test코드에서 Post요청시 인자값을 json형태로 body에 담아서 가도록 변경
- 패스워드 변경 테스트
- Post요청은 @RequestBody를 이용하여 받아오도록 설정

Commit:
회원 주소변경 추가

Commit:
- return값에 불확실한 boolean값 제거
- 리턴타입 enum 상태값 추가
- 아이디 중복체크시 id로 조회한 데이터의 count를 이용하여 체크하도록 변경

yyy9942 committed:
Update README.md

f-lab-dev commented on README.md:
```diff
@@ -1,19 +1,17 @@
 
-### Guides
-The following guides illustrate how to use some features concretely:
+## 프로젝트 화면 설계
```
좀 더 적극적으로 어필하셔도 될 것 같네요~
서버를 집중적으로 하기위해 프론트엔드 개발에 들어가는 공수를 절약하기 위해 프로토타이핑 툴을 사용하여 프론트엔드의 화면을 정의하고, 이를 중심으로 필요한 데이터를 구상했다

f-lab-dev commented on README.md:
```diff
@@ -1,19 +1,17 @@
+## 프로젝트 DB ERD
+
+aquerytool - <http://aquerytool.com:80/aquerymain/index/?rurl=e98cd618-c212-4d0e-ae40-866e1a0cb4b5&> <br>
+Password : yxjr01
```
비밀번호가 써있는건 좋지 않은 것 같네요. readonly 권한만 가진 URL이 있으면 좋을 것 같아요~

f-lab-dev commented on README.md:
```diff
@@ -1,19 +1,17 @@
-* [MyBatis Framework](https://mybatis.org/spring-boot-starter/mybatis-spring-boot-autoconfigure/)
-* [Spring Web](https://docs.spring.io/spring-boot/docs/{bootVersion}/reference/htmlsingle/#boot-features-developing-web-applications)
+## 프로젝트 포스팅
+Tistory - <https://deveric.tistory.com/category/Project/DelFood>   
```
여기도 부가적인 설명이 있으면 좋을 것 같아요. 프로젝트에서 생기는 기술적인 어려움을 해결할 때의 경험을 블로그로 정리했다

f-lab-dev commented on README.md:
```diff
@@ -1,19 +1,17 @@
-# Getting Started
+DelFood
+=============
+전국에 있는 배달음식점과 소비자를 연결하는 중개  플랫폼 서버 입니다.<br>
```
배달의 민족을 벤치마킹했다고 하면 읽는 사람이 더 이해하기 쉬울 것 같습니다~

Commit:
알 수 없는 에러 발생시 Throw Exception 하도록 변경.
MemberDTO에 NotNull 조건 추가
로그인시 MemberDTO가 아니라 MemberLoginRequest로 정보를 가져오도록 변경
그밖에 주석 추가

Commit:
Exception 발생시 사용하지 않는 코드 제거
사용하지 않는 import 제거

Commit:
Google CheckStyle 적용

Commit:
Exception, Redis Session 관련 주석 추가

yyy9942 committed:
Update Readme.md

화면 설계 이미지, ERD 이미지 추가

Commit:
협업을 위한 Owner 기본 Class, Interface 제작

Commit:
Owner SignUp, IdDuplCheck 구현

Commit:
회원가입 시 패스워드 암호화 적용

Commit:
회원 정보 변경 또는 탈퇴시 updated_at이 적용되지 않던 버그 수정

binaryyoung committed:
사장님 기능 구현

- 로그인, 로그아웃
- 사장님 정보 조회
- 사장님 이메일, 전화번호 수정
- 사장님 비밀번호 수정

Commit:
사용하지 않는 OwnerController 상태코드 제거

Commit:
매장 도메인 개발
- 매장 생성 개발

binaryyoung committed:
- @Alias 어노테이션을 이용해 typeAlias를 사용할 수 있도록 변경하였습니다.
- 비밀번호 변경 시 아이디와 기존비밀번호가 일치하지 않으면 UNATHORIZED를 보내도록 수정하였습니다.
- 이메일,주소 수정 시 하나만 null인경우에는 하나만 수정이 되도록 변경하였습니다.
- 수정작업이 정상적으로 진행되거나 아예 일어나지 않은 경우를 제외하고는 롤백처리 되도록 수정하였습니다.

yyy9942 committed:
Merge pull request #9 from f-lab-edu/featurue/1

#8 Featurue/1

yyy9942 committed:
Merge pull request #4 from f-lab-edu/feature/member_join

#6 고객 도메인 관리 개발

Commit:
Merge branch 'develop' into feature/project_structure_fix

Conflicts:
	README.md
	src/main/java/com/delfood/vo/MemberVO.java

yyy9942 committed:
Merge pull request #3 from f-lab-edu/feature/project_structure_fix

코드리뷰를 반영하였습니다!

Commit:
리뷰 내용 반영

- Controller에서 더이상 DMLResponse를 받지 않도록 수정
signUp시 에러발생이 되면 Exception이 발생하니 return값을 받을 필요가 없다고 판단하여 return값을
삭제하였습니다.

- DTO마다 있는 hasNullDataBeforeInsert 메서드의 구조 변경
실제로 return되는 값은 똑같습니다. if문을 제거하였습니다.

binaryyoung committed:
- DB 관련 정보 삭제

- controller의 login 메서드가 에러를 throw하지 않도록 변경

- 사장 정보 수정 시 Patch 방식을 이용하도록 변경

- 메일, 전화번호 수정 메서드에 비밀번호가 맞아야 수정할 수 있도록 변경

- 비밀번호 수정 시 기존 비밀번호와 새 비밀번호 NPE 방지 조건 수정

- 사장 업데이트  Response 클래스 하나로 통합

- OwnerService 로그인 메서드 사장 정보 조회 메서드로 변경

Commit:
Query 오타 수정, Menu를 DTO로 뺌

- Query에 존재하던 오타 수정
- 동적 update Query 추가
- SignatureMenu enum을 제거하고 테이블을 추가 생성하여 DTO로 뺌

yyy9942 committed:
Merge pull request #14 from f-lab-edu/feature/5

#7 Feature/5

yyy9942 committed:
Merge pull request #13 from f-lab-edu/feature/4

#10 Feature/4

Commit:
충돌 해결 및 버그 수정

Merge remote-tracking branch 'yyy9942/feature/2' into feature/3

Conflicts:
	src/main/java/com/delfood/controller/OwnerController.java
	src/main/java/com/delfood/dto/OwnerDTO.java
	src/main/java/com/delfood/mapper/OwnerMapper.java
	src/main/java/com/delfood/service/OwnerService.java
	src/main/resources/mybatis/mapper/owner.xml

	========================

버그 수정
1. xml(sql) 오류로 인한 런타임 에러 수정
2. 로그아웃을 Post요청에서 Get요청으로 변환
3. Shop insert 쿼리 오류 수정

yyy9942 committed:
Merge pull request #12 from f-lab-edu/feature/3

#10 Feature/3

Commit:
DMLOperationError -> DMLOperationResult

Commit:
Member Logout 기능 추가, 오타 수정

Commit:
리뷰 반영
- DMLOperationResult -> OperationResult
- final이 없는 불변객체를 final로 만듦
- newShop -> addShop
- tab -> space

Commit:
SessionUtil 적용
- Member에서 session을 직접 작업하지 않고, SessionUtil을 통해 작업하도록 변경
- Owner에서 session을 직접 작업하지 않고, SessionUtil을 통해 작업하도록 변경

Commit:
매장 조회 기능 추가
1. 페이징 적용
- page 값을 주지 않을시 자동으로 0 값을 주입

2. 총 매장 개수를 같이 return하여 매장 개수를 알 수 있도록 만듦

Commit:
Code 리팩토링 :  불필요한 조건문 제거, 트랜잭션 적용 등
- OperationResult enum 제거
- return Type으로 존재하던 OprationResult는 모두 void로 적용
- DML문에 Transactional 적용
- 예외 발생시 로그를 남기도록 적용
- LIMIT로 인한 성능저하를 피하기 위해 페이징 쿼리 수정

Commit:
Shop Update 추가
- ShopUpdateDTO를 추가하여 정보 업데이트시 가져오는 정보를 표시
- 업데이트 매장이 로그인한 사장님 소유인지 확인

binaryyoung committed:
메뉴그룹 DTO 및 Mapper, Service 생성

yyy9942 committed:
Merge pull request #17 from f-lab-edu/feature/7

#16 Feature/7

Commit:
매장 오픈, 마감기능 추가, 코드리뷰 반영
- 오픈, 마감 시도시 각각 오픈과 마감이 가능한 상태인지 확인한다.
- 오픈, 마감시 각각 로그를 기록한다.
- 오픈시에는 로그를 insert, 마감시에는 로그를 update한다.
* Work ==> 매장 영업 기록

- ShopUpdateDTO 세팅시 값을 복사하여 전달하도록 변경

Commit:
updateDTO에 final 키워드 추가

yyy9942 committed:
Update README.md

binaryyoung committed:
메뉴 관리 기능 구현
- 메뉴관리할 매장 조회
- 매장 별 메뉴 조회
- 메뉴 그룹 추가

Commit:
주소 변경 작업
- Member
1. address, address_detail 삭제
2. address_code 추가, AddressDTO 추가
3. 정보 조회시 resultMap을 이용하여 조회하도록 변경

- Shop
1. address, address_detail 삭제
2. address_code 추가
=> 추후 AddressDTO와 배달지역을 저장하기 위한 List<AddressDTO>를 추가할 예정

binaryyoung committed:
이전 커밋 자료 누락되어 다시 커밋합니다.

메뉴 관리 기능 구현 …

- 메뉴관리할 매장 조회
- 매장 별 메뉴 조회
- 메뉴 그룹 추가

Commit:
매장에 배달 지역 추가, 삭제, 조회 기능 추가
addressCode 말고 상세 주소(xxx호) 정보가 필요하다고 판단하여 해당 컬럼 추가

binaryyoung committed:
- MenuGroupDTO 코드 컨벤션 수정, MenuGroup 수정 기능 추가

Commit:
카테고리, 위치 기반 매장 검색 구현

binaryyoung committed:
코드리뷰 반영 및 메뉴 그룹 삭제 기능 추가
- checkStyle이 적용이 안되있어서 다시 적용
- menu의 모든 내용을 조회하는 공통쿼리 제거

Commit:
리뷰 적용

Commit:
사장 로그인 체크, 매장 주인 체크에 AOP 적용
- AOP 의존성 추가
- AOP 적용 어노테이션 생성
- 중복되는 로그인체크, 매장 주인 체크 로직 삭제
- 어노테이션을 적용하여 로그인체크, 매장 주인체크 진행

Commit:
Member AOP 적용

Commit:
매장 일괄 오픈, 일괄 마감 기능 추가, 매장 검색시 회원 로그인을 하도록 변경

Commit:
주소 검색 로직 개발

binaryyoung committed:
- 메뉴그룹 순서변경 기능 추가
- 에러 처리 관련 기능 추가

binaryyoung committed:
- 메뉴와 메뉴그룹 분리
- 메뉴 생성 기능 추가

Commit:
리뷰 반영
- Aspect에서 CommonResponse 삭제
- HttpStatusCodeException을 사용하여 throw하도록 변경

Commit:
- 리뷰 반영
1. 주소 조회시 필요한 컬럼만 조회하도록 설정
2. AOP 로그인권한 체크 로직을 Around -> Before로 변경
3. CheckStyle 미적용된 부분에 컨밴션 적용

binaryyoung committed:
- 메뉴 조회, 수정, 순서변경, 삭제 기능 추가
- 옵션 추가, 삭제 기능 추가
- 코드리뷰 반영

yyy9942 committed:
Merge pull request #15 from f-lab-edu/feature/6

#6 Feature/6

yyy9942 committed:
Merge pull request #26 from f-lab-edu/feature/5

Feature/5

yyy9942 committed:
Merge pull request #27 from f-lab-edu/feature/4

Feature/4

yyy9942 committed:
Merge pull request #28 from f-lab-edu/feature/3

Feature/3

Commit:
배달지역 추가시 유효성 검증, 자잘한 코드 리팩토링, 주소 관련 로직 추가

1. 배달 지역 추가시 유효성 검증
- 배달지역 추가 데이터를 Set으로 받아와서 중복된 추가 값을 하나로 통일
- 이미 추가되어있는 지역을 추가하려고 시도할 때 유니크 키 익셉션이 발생하는 것을 랩핑하여 예외처리 진행

2. 코드 리팩토링
- 위치정보 관련된 기능이 ShopMapper에 존재해서 DeliveryLocationMapper로 이동시킴
- 그 밖에도 메서드 위치가 잘못된 것을 찾아서 이동시킴
- ResponseEntity에 담을 정보가 없는 컨트롤러 메서드의 반환 값을 void로 변경
- 요청 url을 좀 더 직관적으로 변경
- 자잘한 변수 이름 수정 및 컨밴션 적용

3. 주소 로직
- 자신의 배달 가능 지역을 불러오는 기능 추가
- 배달 가능 지역을 추가할 때 한번에 다중 insert를 진행하도록 변경
- DB에 shop_id와 town_code를 유니크 키로 만들어서 중복된 배달지역이 생기지 않도록 변경
- DB에서 유니크키 관력 Exception이 발생하면 그 Exception을 랩핑하도록 추가
- 자신의 매장 정보를 조회할 때 AdressDTO로 배달 주소 조회 -> DeliveryLocationDTO로 배달
주소조회하도록 변경

binaryyoung committed:
- 다중 insert를 활용한 옵션 리스트 insert 기능 추가
- OptionDTO hasNullDataBeforeCreate()에서 menuId도 체크목록에 추가
- 기타 SQL 문법 및 로그 수정

yyy9942 committed:
Update README.md

Commit:
Redis Caching 설정 및 오타 수정

- 캐싱을 위한 설정및 의존성 추가
- 주소 검색, 배달 가능지역 조회에 시범적으로 캐싱 적용

Commit:
리뷰 반영
- 필요없는 toString 오버라이드 제거
- CacheKeys 상수 클래스를 제거
- 가독성 향상을 위한 개행

Commit:
리뷰 반영
- SessionUtil 메서드 이름 변경

Commit:
ShopController에서 반환하는 객체 중 상수화 시킬 수 있는 객체를 상수화

binaryyoung committed:
- controller에서 단순히 httpStatus.OK를 반환하는 부분 void 반환으로 변경
- update 관련 exception 추가
- entity 관련 exception 패키지 분리
- 코드리뷰 반영

binaryyoung committed:
누락된 exception 추가

Commit:
Redis Connection Factory 주석 수정

Commit:
오타 수정

Commit:
캐싱 관련 로직 수정
- 주소 검색 Request를 도로명, 지번검색으로 나눔
- 캐싱 키를 제작하는 로직 제작
- 캐싱을 서비스 레이어에서 진행하도록 변경

주소 검색 쿼리 수정
- 통합 검색을 도로명, 지번 검색에 따른 쿼리로 나눔
- NULL값이 들어올 경우 공백으로 검색하도록 변경

Commit:
회원 단위테스트 작성

yyy9942 committed:
Merge pull request #31 from f-lab-edu/feature/12

#30 Feature/12

yyy9942 committed:
Merge pull request #29 from f-lab-edu/feature/11

feature/11

yyy9942 committed:
Merge branch 'feature/2' into feature/10

yyy9942 committed:
Merge pull request #24 from f-lab-edu/feature/10

#23 Feature/10

Commit:
conflict 해결 - deleteMemberInfo

binaryyoung committed:
Merge remote-tracking branch 'origin/feature/2' into feature/9

binaryyoung committed:
Merge remote-tracking branch 'origin/feature/2' into feature/9

Conflicts:
	src/main/java/com/delfood/controller/MemberController.java
	src/main/java/com/delfood/mapper/MemberMapper.java
	src/main/java/com/delfood/mapper/ShopMapper.java
	src/main/java/com/delfood/service/ShopService.java
	src/main/resources/mybatis/mapper/shop.xml

binaryyoung committed:
Merge branch 'feature/2' into develop

Conflicts:
	src/main/resources/mybatis/mapper/member.xml

Commit:
conflict 해결

Commit:
MenuGroup 단위테스트 작성
Member 단위테스트에서 사용하지 않는 import 제거

binaryyoung committed:
- Owner 회원가입 테스트 케이스 추가

yyy9942 committed:
Merge pull request #32 from f-lab-edu/feature/13

#30 Frature/13

Commit:
고객 Controller, Service의 로직을 더 명확하게 분리, 고객 단위테스트 코드 수정

Commit:
- 사용하지 않는 내부클래스 제거
- Exception handler 추가
- 사장님, 매장의 컨트롤러와 서비스 로직을 더 명확하게 분리
- 코드 리팩토링

Commit:
통합테스트 주석처리

Commit:
properties 배포

binaryyoung committed:
Merge pull request #33 from f-lab-edu/feature/14

#30 feature/14

binaryyoung committed:
메뉴 그룹 코드 리팩토링
- 컨트롤러에 내부 클래스로 만든 response 일부 삭제
- 컨트롤러에 유효성 체크를 하는 부분을 제외한 비지니스 로직 서비스로 이동

Commit:
리뷰반영 및 컨트롤러 리팩토링

open shop때는 shop의 정보를 넘겨주었지만 close할때는 정보를 넘겨주지 않던 것을 close할때도 정보를 넘겨주도록
변경

Commit:
리뷰 반영
- id 중복체크 url을 명시적으로 변경

binaryyoung committed:
#### MenuGroupController
- MenuGroup enum 대신 String 상수로 변경
- AOP 어노테이션 적용

#### MenuGroupService
- 세분화된 중복 Exception 삭제
- 파라미터 값 변경 부분 전환

#### MenuController
- 메뉴 조회시 httpStatus.OK도 같이 보낼 수 있도록 반환타입을 ResponseEntity로 변경

#### MenuService
- addMenu() : 리턴할 메뉴아이디가 없는 경우 롤백되도록 변경
- checkMenu() : 예외처리 부분을 boolean 반환하도록 변경

Commit:
메뉴, 옵션 단위테스트 작성

Commit:
리뷰 반영
- import * 제거
- 비슷한 테스트 통합

나머지 내용은 본 로직이 수정되어야 해서  코드 수정해달라고 요청하였습니다.

yyy9942 committed:
Merge pull request #34 from f-lab-edu/feature/15

#16 #30 feature/15

Commit:
장바구니 기능 구현
- DAO는 redis를 사용하도록 구성
- 고객이 메뉴를 장바구니에 넣을 시 해당 현제 메뉴가 해당 매장의 메뉴인지 확인한 후 입력

binaryyoung committed:
리뷰 반영 및 기존 코드 리팩토링

Commit:
cart index delete 기능 구현 및 Test code 작성

binaryyoung committed:
Owner 테스트 코드 작성 및 리팩토링

Commit:
- Redis tranasaction 적용
- redis에 '@transactional'을 적용할 수 있도록 트랜잭션 매니저 빈 등록
- Key를 enum으로 관리
- 리뷰 반영

binaryyoung committed:
- 코드리뷰 반영
- ShopServiceTest addShop() 테스트 코드 추가

Commit:
장바구니 로직 추가

- 총 가격 계산
- 동일 메뉴를 넣으려할때 버그 수정
- 장바구니 최대 개수 10개로 수정

binaryyoung committed:
- OrderDTO 생성
- 주문 신청 기본 정보 조회 기능 추가
- 임시로 CartService 추가

Commit:
- 캐싱하기 적합한 DTO를 사용하도록 변경
- 가격 계산로직을 캐싱에 의존하도록 변경
- 테스트코드 삭제(로직 완성시 다시 작성할 예정)

Commit:
빈 테스트코드 추가(Test ERROR)

yyy9942 committed:
Merge pull request #39 from f-lab-edu/feature/16

#38 Feature/16

yyy9942 committed:
Merge pull request #47 from f-lab-edu/feature/15

Feature/15

binaryyoung committed:
Merge pull request #43 from f-lab-edu/feature/19

#42 #30 feature/19

binaryyoung committed:
OrderDTO에 @JsonFormat 추가

Commit:
리뷰 반영
- 장바구니 가격 계산 로직 분리
- redis 트랜잭션에서 discard를 명시적으로 선언
- redis '@Transactional' 삭제
- 사용하지 않는 세션용 변수 제거
- ItemDTO를 불변객체로 지정

binaryyoung committed:
- 쿠폰 추가 기능 생성

Commit:
사용하지 않는 menuService 제거
리뷰 반영

Commit:
DTO를 Inner Class로 분리

Commit:
가상 결제 구현

yyy9942 committed:
Update README.md

yyy9942 committed:
Update README.md

Commit:
사용하지 않는 코드 제거

binaryyoung committed:
- 메뉴관련 xml 파일 체크스타일 적용
- 메뉴그룹 순서 조정 메서드 수정으로 인한 테스트 코드 변경
- DTO Null 체크 메소드 변경

Commit:
주문 로직 개발
- 주문 요청 전 가격 계산
- 주문 요청시 전처리 실행

binaryyoung committed:
코드리뷰 반영 및 쿠폰 수정 삭제 기능 추가

Commit:
계산서 발행 로직 작성 및 리뷰 일부 반영

Commit:
거리 계산로직 추가

Commit:
배달료 계산 로직 추가

binaryyoung committed:
- 코드리뷰 반영
- 쿠폰발급, 발급쿠폰 사용기능 추가

binaryyoung committed:
- 발급 쿠폰 조회기능 추가 및 CouponIssueDTO 변경
- 주석 빠진 부분 추가
- 로직만 작성했던 부분을 postman 테스트하면서 잘못 설정된 부분을 고쳤습니다.

Commit:
insert문을 벌크 insert로 진행하도록 변경

Commit:
주문 기능 제작
- 주문 조회시 쿼리 한번으로 다 조회하도록 코딩
- 나머지 쿼리문제는 차차 해결해야함
- 아직 삽질중이라 작동은 하는데 많이 느림

Commit:
계산서 발행 시 메뉴, 메뉴 옵션에 의해 다수의 SELECT 쿼리가 질의되는 문제 해결

Commit:
자신의 주문 정보 조회 기능 추가
주문 아이디로 주문 상세조회 추가
자신의 주문인지 검증하는 로직 만들어야함
페이징도 만들어야함

binaryyoung committed:
Merge remote-tracking branch 'origin/feature/17' into develop

Conflicts:
	src/main/java/com/delfood/error/ErrorController.java
	src/main/java/com/delfood/error/exception/DuplicateException.java

binaryyoung committed:
- duplicationException 재 커밋

Commit:
OrderDTO 불변객체로 변경
Builder 패턴을 사용하여 만들도록 변경
사용하지 않는 서비스 제거

Commit:
유효성 검사 추가 및 변경
- 주문 전 유효성 검사 추가
- 유효성 검사를 Controller에서 담당하도록 코드 이동

Commit:
OrderDTO 수정으로 인한 버그 수정

binaryyoung committed:
- shopService 불필요한 import 제거
- shopService 테스트 케이스 추가

binaryyoung committed:
-shopService 메서드 가독성있게 변경
notOpenCheck -> isClose

yyy9942 committed:
Update README.md

Commit:
리뷰 반영

binaryyoung committed:
shopService 메서드 명 변경에 따라 Test 메서드 이름 변경 (notOpenCheck() -> isClose())

Commit:
가상 결제 시스템 브랜치와 병합
회원 주문 조회시 페이징 처리 추가
주문번호로 주문 조회시 유효성 검사 추가

Commit:
가상 결제 진행 개발

Commit:
FCM Push 서비스 제작.

Commit:
토큰 추가 기능 제작
메서드 이름을 직관적으로 변경

binaryyoung committed:
-shopService isClose() 메서드 사용 부분 가독성있게 변경
-WorkService 테스트 코드 추가

Commit:
리뷰 반영
- 동시성 이슈를 처리하기 위한 트랜잭션 적용
- 명시적 변수 이름 변경

Commit:
주문 조회시 없는 주문 번호로 인한 버그 발생 수정

binaryyoung committed:
Merge pull request #54 from f-lab-edu/feature/23

[#53] Shop, Address, work 테스트 코드 작성

yyy9942 committed:
Merge pull request #52 from f-lab-edu/feature/22

[#50] mock 결제 서비스 제작

Commit:
리뷰 반영
- 예외에 오류가 난 정보를 출력하도록 변경
- 적절한 예외로 변경

Commit:
푸시메세지 비동기 전송 방식으로 개선

Commit:
주문시 사장님에게 푸시 메세지로 알림.
푸시 적용할 곳이 아직 없네요.

Commit:
AddressService 테스트코드 작성

Commit:
장바구니 서비스 대상 테스트코드 작성

yyy9942 committed:
Merge pull request #58 from f-lab-edu/주소_서비스_테스트_코드_작성

[#53] 주소 서비스 테스트 코드 작성

Commit:
팩토리 메서드 개선

Commit:
리뷰 반영
- 푸시 메세지 전송시 에러가 발생하면 예외를 발생시키지 않음. 대신 로그에 해당 정보를 기록하고 redis에 해당 오류가 난
정보를 기록하여 추후 다시 푸시메세지 전송을 시도할 수 있도록 함

Commit:
Merge branch '푸시_서비스_적용' of https://github.com/f-lab-edu/food-delivery.git into 푸시_서비스_적용

Commit:
버그, 오류 수정
- 주문 INSERT시 배달료가 DB에 입력되지 않던 버그 수정
- 푸시 토큰이 0개이면 예외 발생하던 것을 로그만 찍어주도록 변경

yyy9942 committed:
Update README.md

yyy9942 committed:
Merge pull request #60 from f-lab-edu/장바구니_서비스_테스트_코드_작성

[#59] 장바구니 서비스 대상 테스트코드 작성

yyy9942 committed:
Merge pull request #63 from f-lab-edu/주소_서비스_테스트_코드_작성

주소 서비스 테스트 코드 작성

Commit:
Merge remote-tracking branch 'yyy9942/feature/21' into 푸시_서비스_적용

Conflicts:
	pom.xml
	src/main/java/com/delfood/dto/ShopDTO.java
	src/main/java/com/delfood/error/ErrorController.java

Commit:
Coupon에 대한 테스트코드 작성

Commit:
계산시 쿠폰 적용 기능 추가

Commit:
- 내 주문 조회에서도 쿠폰 적용한 결과를 볼 수 있도록 수정
- 자잘한 버그 수정

Commit:
리뷰 반영
- 발행 쿠폰 조회시 NULL 체크
- 로그 하나로 통합

Commit:
sql 버그 수정
- coupon이 'cooupon'으로 오타가 났던 것을 수정

주문성공시 상태 변경
- 결제 완료시 order_status가 'BEFOR_PAYMENT'에서 'ORDER_REQUEST'로 변경되도록 설정함

주문이 중간에 실패하더라도 기록이 남도록 변경
- 트랜잭션 경계 설정을 통해 실패시 기록을 남기도록 설정

Commit:
주문 승인 기능 개발
- 주문 승인 시 도착예정시간 설정 가능
- 주문 승인 시 고객에게 알림 메세지 전송

yyy9942 committed:
Update README.md

Commit:
리뷰 반영
- 메서드 이름 변경

Commit:
라이더 정보 서비스 개발
- 회원 가입
- 로그인
- 로그아웃
- 아이디와 비밀번호로 유효성 검사
- 비밀번호 변경
- 메일 변경

yyy9942 committed:
Merge pull request #65 from f-lab-edu/order_coupon

[#46] 주문시 쿠폰 적용

yyy9942 committed:
Merge pull request #64 from f-lab-edu/coupon_test

Coupon에 대한 테스트코드 작성

Commit:
리뷰 반영
- 어노테이션 개선
- 리플랙션을 활용하여 어노테이션 내부 로직 변경

Commit:
배달 매칭 서비스 제작

Commit:
배달 매칭 제작

yyy9942 committed:
Update README.md

yyy9942 committed:
Update README.md

yyy9942 committed:
Merge pull request #68 from f-lab-edu/rider_info_service

[#67] 라이더 회원 정보 서비스

yyy9942 committed:
Merge pull request #71 from f-lab-edu/order_coupon

Order coupon

yyy9942 committed:
Merge branch 'develop' into feature/20

yyy9942 committed:
Merge pull request #46 from f-lab-edu/feature/20

[#45] 주문기능 제작

yyy9942 committed:
리뷰 적용

yyy9942 committed:
푸시_서비스_적용에 적용한 리뷰를 Merge, 해당 내용으로 발생하는 오류 수정

yyy9942 committed:
배달원 매칭 개발

yyy9942 committed:
Merge pull request #57 from f-lab-edu/푸시_서비스_적용

[#55] 푸시 서비스 적용

yyy9942 committed:
Merge pull request #41 from f-lab-edu/feature/18

[#40] 장바구니 기능 구현

yyy9942 committed:
리뷰 반영

Commit:
maven에서 빌드되지 않는 버그 수정

yyy9942 committed:
리뷰  반영

yyy9942 committed:
배달원의 배달 관련 기능 추가
- 배달 대기명단에 올리기
- 배달 대기명단 삭제
- 내 배달 현황
- 내가 배달한 물품들 (페이징 포함)

yyy9942 committed:
리뷰반영

Commit:
오늘의 배달 수입 조회 기능 추가

Commit:
버그 수정

yyy9942 committed:
- 리뷰 반영
-  라이더 매칭이 제대로 되지 않는 버그 수정
- ConcurrentHashMap이 멈추는 현상 수정

yyy9942 committed:
위치 관련 버그 수정

yyy9942 committed:
주문테스트 일부 제작

yyy9942 committed:
'@Autowired'관련 버그 수정, 오타 수정

yyy9942 committed:
주문 테스트 작성

yyy9942 committed:
가상결제 테스트 작성

yyy9942 committed:
패키지 구조 변경

Commit:
배달 매칭 RedisDao 구현

Commit:
테스트코드 작성

Commit:
라이더 회원 정보 서비스 테스트코드 작성

yyy9942 committed:
release 프로파일 추가

yyy9942 committed:
Build 버그 수정

yyy9942 committed:
누락 정보 수정

yyy9942 committed:
오탈자 수정

yyy9942 committed:
.

yyy9942 committed:
redis 호스트와 포트 적용 되지 않는 버그 수정

yyy9942 committed:
Update README.md

yyy9942 committed:
Merge pull request #78 from f-lab-edu/feature/rider_info_service_test

[#77] 라이더 회원정보 서비스 테스트코드 작성

yyy9942 committed:
jenkins push를 위한 commit

yyy9942 committed:
Update README.md

Commit:
Jenkins 빌드 유발

Commit:
Merge remote-tracking branch 'yyy9942/feature/delivery_dao_redis' into
feature/release_test

Conflicts:
	src/main/java/com/delfood/dao/deliveery/DeliveryDao.java
	src/main/java/com/delfood/dao/deliveery/LocalMemoryDeliveryDao.java
	src/main/java/com/delfood/service/delivery/DeliveryService.java
	src/main/resources/application-dev.properties

Commit:
오류 해결

Commit:
Merge remote-tracking branch 'yyy9942/feature/release_test' into develop

Conflicts:
	src/main/java/com/delfood/controller/MenuGroupController.java
	src/main/java/com/delfood/dto/MenuDTO.java
	src/main/java/com/delfood/dto/OptionDTO.java
	src/main/java/com/delfood/dto/ShopDTO.java
	src/main/java/com/delfood/error/ErrorController.java
	src/main/java/com/delfood/mapper/MenuMapper.java
	src/main/java/com/delfood/service/MenuGroupService.java
	src/main/java/com/delfood/service/ShopService.java
	src/main/resources/mybatis/mapper/menu.xml
	src/main/resources/mybatis/mapper/menuGroup.xml
	src/test/java/com/delfood/service/MenuGroupServiceTest.java
	src/test/java/com/delfood/service/MenuServiceTest.java
	src/test/java/com/delfood/service/ShopServiceTest.java

Commit:
Merge branch 'feature/release_test' into develop

Conflicts:
	src/main/java/com/delfood/controller/MenuGroupController.java
	src/main/java/com/delfood/dto/MenuDTO.java
	src/main/java/com/delfood/dto/OptionDTO.java
	src/main/java/com/delfood/dto/ShopDTO.java
	src/main/java/com/delfood/mapper/MenuMapper.java
	src/main/java/com/delfood/service/MenuGroupService.java
	src/main/java/com/delfood/service/ShopService.java
	src/main/resources/mybatis/mapper/menu.xml
	src/main/resources/mybatis/mapper/menuGroup.xml
	src/test/java/com/delfood/service/MenuGroupServiceTest.java
	src/test/java/com/delfood/service/MenuServiceTest.java
	src/test/java/com/delfood/service/ShopServiceTest.java

Commit:
properties 추가

Commit:
오류 수정
